modular_skyrat/modules/primitive_cooking_additions/code/cauldron.dm

```diff
@@ -0,0 +1,339 @@
+	eject()
+	return ..()
+
+/obj/machinery/cauldron/Destroy()
+	QDEL_LIST(ingredients)
+	return ..()
+
+/obj/machinery/cauldron/add_context(atom/source, list/context, obj/item/held_item, mob/user)
+	. = ..()
+	if(held_item?.tool_behaviour == TOOL_WRENCH)
+		context[SCREENTIP_CONTEXT_LMB] = "[anchored ? "Unsecure" : "Secure"]"
+		return CONTEXTUAL_SCREENTIP_SET
+
+	if(held_item?.tool_behaviour == TOOL_CROWBAR)
+		context[SCREENTIP_CONTEXT_LMB] = "Deconstruct"
+		return CONTEXTUAL_SCREENTIP_SET
+
+	context[SCREENTIP_CONTEXT_LMB] = "Show menu"
+
+	if(length(ingredients) != 0)
+		context[SCREENTIP_CONTEXT_RMB] = "Start cooking"
+
+	return CONTEXTUAL_SCREENTIP_SET
+
+#define CAULDRON_INGREDIENT_OVERLAY_SIZE 14
+
+/obj/machinery/cauldron/update_overlays()
+	. = ..()
+
+	var/ingredient_count = 0
+
+	for(var/atom/movable/ingredient as anything in ingredients)
+		var/image/ingredient_overlay = image(ingredient, src)
+
+		var/list/icon_dimensions = get_icon_dimensions(ingredient.icon)
+		ingredient_overlay.transform = ingredient_overlay.transform.Scale(
+			CAULDRON_INGREDIENT_OVERLAY_SIZE / icon_dimensions["width"],
+			CAULDRON_INGREDIENT_OVERLAY_SIZE / icon_dimensions["height"],
+		)
+
+		ingredient_overlay.pixel_x = ingredient_shifts_x[(ingredient_count % ingredient_shifts_x.len) + 1]
+		ingredient_overlay.pixel_y = ingredient_shifts_y[(ingredient_count % ingredient_shifts_y.len) + 1]
+		ingredient_overlay.layer = FLOAT_LAYER
+		ingredient_overlay.plane = FLOAT_PLANE
+		ingredient_overlay.blend_mode = BLEND_INSET_OVERLAY
+
+		ingredient_count += 1
+
+		. += ingredient_overlay
+
+	var/base_icon_state = "cauldron_front"
+	var/lid_icon_state
+
+	if(open)
+		lid_icon_state = "cauldron_lid_open"
+	else
+		lid_icon_state = "cauldron_lid_closed"
+
+
+	. += mutable_appearance(
+		icon,
+		lid_icon_state,
+	)
+
+	. += base_icon_state
+
+#undef CAULDRON_INGREDIENT_OVERLAY_SIZE
+
+/obj/machinery/cauldron/update_icon_state()
+	if(operating)
+		icon_state = "cauldron_back_cooking"
+	else
+		icon_state = "cauldron_back_off"
+
+	return ..()
+
+/obj/machinery/cauldron/wrench_act(mob/living/user, obj/item/tool)
+	if(default_unfasten_wrench(user, tool))
+		update_appearance()
+	return ITEM_INTERACT_SUCCESS
+
+/obj/machinery/cauldron/crowbar_act(mob/living/user, obj/item/tool)
+	user.balloon_alert_to_viewers("disassembling...")
+	if(!tool.use_tool(src, user, 2 SECONDS, volume = 100))
+		return
+	new /obj/item/stack/sheet/mineral/stone(drop_location(), 5) // Made with stone instead of iron so that it doesn't outbalance microwaves on station
+	deconstruct(TRUE)
+	return ITEM_INTERACT_SUCCESS
+
+/obj/machinery/cauldron/attackby(obj/item/item, mob/living/user, params)
+	if(operating)
+		return
+
+	if(!anchored)
+		if(IS_EDIBLE(item))
+			balloon_alert(user, "not secured!")
+			return TRUE
+		return ..()
+
+	if(istype(item, /obj/item/storage))
+		var/obj/item/storage/tray = item
+		var/loaded = 0
+
+		if(!istype(item, /obj/item/storage/bag/tray))
```

Comment:
tray/storage dumping doesnt check for item weight class, if you do move it here have it be accessed from a property on the cauldron, or centralize the sanity logic prefferably.